File: CafeEase/CafeEase.Model/Requests/UserInsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace CafeEase.Model.Requests
{
    public class UserInsertRequest
    {
        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [MinLength(4)]
        [Required]
        public string Password { get; set; }

        [Required]
        public string Username { get; set; }

        [Required]
        [Compare(nameof(Password), ErrorMessage = "Passwords do not match.")]
        public string PasswordConfirmation { get; set; }

        [Required]
        public int RoleId { get; set; }
    }
}

File: CafeEase/UI/cafeease_mobile/lib/models/user_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_insert_request.g.dart';

@JsonSerializable()
class UserInsertRequest {
  String? firstName;
  String? lastName;
  String? email;
  String? username;
  String? password;
  String? passwordConfirmation;
  int? roleId;
  int? cityId;

  UserInsertRequest({
    this.firstName,
    this.lastName,
    this.email,
    this.username,
    this.password,
    this.passwordConfirmation,
    this.roleId,
    this.cityId
  });

  factory UserInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$UserInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$UserInsertRequestToJson(this);
}
